### ISSUE TITLE
struct-out #:mutable multiple bindings

### ISSUE THREAD

**Issue by bennn** [racket/racket#2123]
On Racket v6.90.0.30, `struct-out` says there are 2 bindings from an identifier when:
- one module exports a struct with a mutable field,
- two modules import the struct and define sub-structs
- a fourth module imports the sub-structs and attempts to provide one via `struct-out`

Reported by `@jerome` on slack (@euhmeuh ?)

Example:
```
#lang racket/base

(module private racket/base
  (struct container ([elements #:mutable]))
  (provide (struct-out container)))

(module sub1 racket/base
  (require (submod ".." private))
  (struct sub-container-1 container ())
  (provide (struct-out sub-container-1)))

(module sub2 racket/base
  (require (submod ".." private))
  (struct sub-container-2 container ())
  (provide (struct-out sub-container-2)))

(require 'sub1 'sub2)
(provide (struct-out sub-container-1))
;; struct-out: multiple bindings (set-container-elements! and set-container-elements!) for structure-type identifier
```

**Comment by bennn** [comment]
I'm feeling lucky #2124

**Comment by mflatt** [comment]
Closes by #2124
